x-pack/plugin/core/src/main/resources/fleet-files.json

```diff
@@ -10,8 +10,8 @@
   },
   "template": {
     "settings": {
-      "index.lifecycle.name": ".fleet-files-ilm-policy",
-      "index.auto_expand_replicas": "0-1"
+      "index.auto_expand_replicas": "0-1",
```

Comment:
I think this will not have effect for deployments updating to the version where this change is released. 
See https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/IndexTemplateRegistry.java#L296

You'll have to bump the template version in `FleetTemplateRegistry` https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/FleetTemplateRegistry.java#L27
and please also document the change next to the new version
e.g. https://github.com/elastic/elasticsearch/blob/3692354284a98d171717d9f2a2e1311b51ea12f3/x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryTemplateRegistry.java#L38


Please also test the branch as a standalone elasticsearch run and upgrade from a previous version to this branch to see that the templates and your code behaves as expected.

Reply:
These templates were added in 8.7 so nobody should have these changes yet, so is this required?

Reply:
Ah gotcha, then we should be fine here

Reply:
My understanding is that there are no such deployments, since these templates are being newly introduced in 8.7: https://github.com/elastic/elasticsearch/pull/91413

Reply:
That PR seems so distant, I can't believe it's targeting `8.7` and not an older version :)

Reply:
Ha it sat blocked on another team's effort for about a month if I recall. This file upload feature was slotted for 8.6 initially but was eventually punted, with a bulk of the work landing early on in the 8.7 development cycle.